@here/harp-examples/src/getting-started_free-camera.ts

```diff
@@ -139,20 +146,54 @@ export namespace FreeCameraAppDebuggingToolExample {
                 pointOfView,
                 this.mapView.canvas
             );
-            transformControls.attach(this.mapView.camera);
+            transformControls.setSpace("world");
+            transformControls.attach(cameraRelativeToEye);
+
+            const applyTransformControls = () => {
+                // Apply helper camera offset to main (map view) camera.
+                this.mapView.camera.position.add(cameraRelativeToEye.position);
+                // Make sure that pitch limit contraint is preserved
+                const ypr = MapViewUtils.extractYawPitchRoll(
+                    cameraRelativeToEye.quaternion,
+                    this.mapView.projection.type
+                );
+                ypr.pitch = Math.max(
```

Comment:
For future: we've got `MathUtills.clamp` in `@here/harp-utils`.

Reply:
Yeah, I have used it even :).